tests/integration/ha/test_horizontal_scaling.py

```diff
@@ -257,3 +274,60 @@ async def test_safe_scale_down_roles_reassigning(
 
     # continuous writes checks
     await assert_continuous_writes_consistency(c_writes)
+
+
+async def test_safe_scale_down_remove_leaders(
+    ops_test: OpsTest, c_writes: ContinuousWrites, c_writes_runner
+) -> None:
+    """Tests the removal of specific units (elected cluster_manager, juju leader).
+
+    The goal of this test is to make sure that:
+     - the CM reelection happens successfully.
+     - the leader-elected event gets triggered successfully and
+        leadership related events on the charm work correctly, i.e: roles reassigning
+    """
+    app = await app_name(ops_test)
+
+    # scale up by 2 units
+    await ops_test.model.applications[app].add_unit(count=2)
+    await ops_test.model.wait_for_idle(
+        apps=[app], status="active", timeout=1000, wait_for_exact_units=5, idle_period=60
+    )
+
+    # scale down: remove the juju leader
+    leader_unit_id = await get_leader_unit_id(ops_test)
+
+    await ops_test.model.applications[app].destroy_unit(f"{app}/{leader_unit_id}")
+    await ops_test.model.wait_for_idle(
+        apps=[app], status="active", timeout=1000, wait_for_exact_units=4, idle_period=120
+    )
+
+    # make sure the duties supposed to be done by the departing leader are done
+    # we expect to have 3 cm-eligible and 1 data-only nodes
+    leader_unit_ip = await get_leader_unit_ip(ops_test)
+    nodes = await all_nodes(ops_test, leader_unit_ip)
+    assert ClusterTopology.nodes_count_by_role(nodes)["cluster_manager"] == 3
+    assert ClusterTopology.nodes_count_by_role(nodes)["data"] == 4
+
+    # scale-down: remove the current elected CM
+    first_elected_cm_unit_id = await get_elected_cm_unit_id(ops_test, leader_unit_ip)
+    assert first_elected_cm_unit_id != -1
+
+    # scale-down: remove the elected cm
+    await ops_test.model.applications[app].destroy_unit(f"{app}/{first_elected_cm_unit_id}")
+    await ops_test.model.wait_for_idle(
+        apps=[app], status="active", timeout=1000, wait_for_exact_units=3, idle_period=90
+    )
```

Comment:
nit

```suggestion
    # scale-down: remove the current elected CM
    first_elected_cm_unit_id = await get_elected_cm_unit_id(ops_test, leader_unit_ip)
    assert first_elected_cm_unit_id != -1
    await ops_test.model.applications[app].destroy_unit(f"{app}/{first_elected_cm_unit_id}")
    await ops_test.model.wait_for_idle(
        apps=[app], status="active", timeout=1000, wait_for_exact_units=3, idle_period=90
    )
```

Reply:
also is there a meaning behind `idle_period=90`

Reply:
hmm I wonder if there is a constant that could be used for `90` to emphasize this - but if this is a bad suggestion plz ignore